### PR TITLE
[CI] Force usage of f-strings

### PR DESCRIPTION
**Pull request by mx-psi**
### What does this PR do?

Add a Python linter to force people to use f-strings as the preferred string formatting option.
Change existing usages to f-strings with the help of `flynt`, and skip some of the more troublesome refactors.

### Motivation

- Ensure consistent string formatting usage across all codebase
- Ensure linters can detect unused or undefined variables (which are detected reliably, unlike dictionary keys)

### Additional Notes

I divided the code rewriting into the automatic and manual parts; please look more carefully into the manual changes.

### Possible Drawbacks / Trade-offs

1. The automatic change is pretty simple and the `flynt` tool seems to be widely used, so I wouldn't worry that much about these.
2. We may find that using `format` is necessary in some cases (e.g. when formatting raw strings or repeating the same arg several times). We can skip the linter in those places but I would expect them to be rare.
3. This bumps the Python requirement **for tooling** to Python 3.7 or later (2018); I think that should be fine.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] The appropriate `team/..` label has been applied, if known.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: tasks/trace_agent.py [hunk 1]
```diff
@@ -82,21 +75,18 @@ def integration_tests(ctx, install_deps=False, race=False, remote_docker=False,
...
     # since Go 1.13, the -exec flag of go test could add some parameters such as -test.timeout
     # to the call, we don't want them because while calling invoke below, invoke
     # thinks that the parameters are for it to interpret.
     # we're calling an intermediate script which only pass the binary name to the invoke task.
     if remote_docker:
-        test_args["exec_opts"] = f"-exec \"{os.getcwd()}/test/integration/dockerize_tests.sh\""
+        exec_opts = f"-exec \"{os.getcwd()}/test/integration/dockerize_tests.sh\""
 
-    go_cmd = 'INTEGRATION=yes go test -mod={go_mod} {race_opt} -v'.format(**test_args)
+    go_cmd = f'INTEGRATION=yes go test -mod={go_mod} {race_opt} -v -tags "{go_build_tags}" {exec_opts}'
```
Comment (mx-psi) [inline tasks/trace_agent.py:89]: This was missing the build tags and exec opts

File: pkg/config/legacy/tests/config.py [hunk 2]
```diff
@@ -327,7 +327,7 @@ def get_config(options=None):
             listen_port = 17123
             if config.has_option('Main', 'listen_port'):
                 listen_port = int(config.get('Main', 'listen_port'))
-            agentConfig['dd_url'] = "http://{}:{}".format(agentConfig['bind_host'], listen_port)
+            agentConfig['dd_url'] = f"http://{agentConfig['bind_host']}:{listen_port}"
```
Comment (albertvaka) [inline pkg/config/legacy/tests/config.py:330]:
```suggestion
            bind_host = agentConfig['bind_host']
            agentConfig['dd_url'] = f"http://{bind_host}:{listen_port}"
```

File: tasks/release.py [hunk 3]
```diff
@@ -718,14 +679,14 @@ def _update_release_json_entry(
     import requests
 
     jmxfetch = requests.get(
-        "https://oss.sonatype.org/service/local/repositories/releases/content/com/datadoghq/jmxfetch/{0}/jmxfetch-{0}-jar-with-dependencies.jar".format(
+        "https://oss.sonatype.org/service/local/repositories/releases/content/com/datadoghq/jmxfetch/{0}/jmxfetch-{0}-jar-with-dependencies.jar".format(  # noqa: FS002
             jmxfetch_version,
         )
```
Comment (albertvaka) [inline tasks/release.py:684]:
```suggestion
        f"https://oss.sonatype.org/service/local/repositories/releases/content/com/datadoghq/jmxfetch/{jmxfetch_version}/jmxfetch-{jmxfetch_version}-jar-with-dependencies.jar"
```
Unless I missed some reason not to change it.

**Review by albertvaka — APPROVED** [review]

File: tasks/release.py (hunk 3 above)
Comment (mx-psi) [reply tasks/release.py:684]: no reason, just that it was a looong line 😄

**Review by brycekahle — APPROVED** [review]

**Review by alai97 — APPROVED** [review]
Looks good for docs!

**Review by djmitche — APPROVED** [review]

**Review by paulcacheux — APPROVED** [review]

**Review by davidor — APPROVED** [review]

**Pull request merged**